cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/SourceControlManagementService.java

```diff
@@ -229,8 +236,15 @@ public ApplicationRecord pullAndDeploy(ApplicationReference appRef) throws Excep
     AppRequest<?> appRequest = pullResponse.getAppRequest();
     SourceControlMeta sourceControlMeta = new SourceControlMeta(pullResponse.getApplicationFileHash());
     // Deploy with a generated uuid
-    String versionId = RunIds.generate().getId();
-    ApplicationWithPrograms app = appLifecycleService.deployApp(appRef.app(versionId), appRequest,
+    ApplicationId appId = appRef.app(RunIds.generate().getId());
+    
+    accessEnforcer.enforce(appId, authenticationContext.getPrincipal(), StandardPermission.CREATE);
```

Comment:
similar question here, do we need a read permission on the repo?

Reply:
Will add the Repository Pull, Push and List permission in a separate PR